src/library_wasm_worker.js

```diff
@@ -30,28 +30,28 @@
 
 
 mergeInto(LibraryManager.library, {
-  wasm_workers: {},
-  wasm_workers_id: 1,
+  $wasmWorkers: {},
```

Comment:
Did we come to consensus on how to annotate internal vs public JS library file symbols?

I don't like this change, because it signals to end users that `wasmWorkers` would be a public symbol they would be able to access in their JS library code and external JS. The previous `_wasm_workers` symbol had the benefit of generating an underscore to signify that this symbol is for internal use, and not available for user JS libraries to access.

Compare this to e.g. `$GLctx`, which is a public symbol and accessible via `GLctx` from user JS code.

Reply:
For symbols that we want to keep internal we can do:

```
$wasmWorkers__internal: true,
```

This prevents user-js libraries from being able to see this symbol.   That maybe isn't always ideal so perhaps we could use some other mangling.

Using the leading underscore to mean "internal" is confusing because all native symbols get that mangling (e.g. `_malloc` and `_free` show up like that in JS code and are **not** internal symbols).

If we do want to stick with underscore as a hint of internal-ness I guess we could do `$_wasmWorkers` ? 

The important thing here is that these symbols need to be excluded from the native code link (they are not visiable to wasm-ld).   This is required for the auto-signature generation to work (at least today) since we assume that all non-$-prefixed symbols can be called from native code.

Reply:
> Using the leading underscore to mean "internal" is confusing because all native symbols get that mangling (e.g. _malloc and _free show up like that in JS code and are not internal symbols).

That is true and I agree it is a bit confusing, but there are extremely few of those functions, `_malloc` and `_free` represent select few examples. This mangling happens mostly for user's own code on both sides. Even if it might not be that `_foo` would always mean private, the opposing direction has held true in most cases: a symbol `foo` reads like it is intended to be public.

In some cases to be extra clear about this, I had used an extra underscore in the JS name, e.g. `_foo` which would expand to `__foo`, or even double underscore (`__call_main` -> `___call_main`).

What is the benefit of using `$_wasmWorkers` over just `wasmWorkers`? Are all `$` symbols excluded from native code link?

Reply:
Yes, symbols all symbols that start with `$` are essentially not visible to native code.   When we generate the set of all symbols that that native code can link against we exclude these "only visible to JS" symbols.

See https://github.com/emscripten-core/emscripten/blob/9dff99d6a303c7987fe195a53c883b1c1b390b70/src/jsifier.js#L33-L37

In my mind this is that primary reason to tag things with `$` but I could be missing some original intention?

The more we tag with `$` the smaller we make the visible API surface that we present to wasm-ld, and the less likely it is that somebody could accidentally link against a JS-only symbol.